src/main/java/org/alliancegenome/curation_api/services/validation/dto/Gff3DtoValidator.java

```diff
@@ -243,6 +254,103 @@
+		TranscriptCodingSequenceAssociation association = new TranscriptCodingSequenceAssociation();
+		
+		ObjectResponse<TranscriptCodingSequenceAssociation> associationResponse = new ObjectResponse<TranscriptCodingSequenceAssociation>();
+		if (!attributes.containsKey("Parent")) {
+			associationResponse.addErrorMessage("Attributes - Parent", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			SearchResponse<Transcript> parentTranscriptSearch = transcriptDAO.findByField("modInternalId", attributes.get("Parent"));
+			if (parentTranscriptSearch == null || parentTranscriptSearch.getSingleResult() == null) {
+				associationResponse.addErrorMessage("Attributes - Parent",ValidationConstants.INVALID_MESSAGE + " (" + attributes.get("Parent") + ")");
+			} else {
+				Transcript parentTranscript = parentTranscriptSearch.getSingleResult();				Map<String, Object> params = new HashMap<>();
+				params.put(EntityFieldConstants.TRANSCRIPT_ASSOCIATION_SUBJECT + ".id", parentTranscript.getId());
+				params.put("transcriptCodingSequenceAssociationObject.id", cds.getId());
+				SearchResponse<TranscriptCodingSequenceAssociation> searchResponse = transcriptCdsDAO.findByParams(params);
+				if (searchResponse != null && searchResponse.getSingleResult() != null) {
+					association = searchResponse.getSingleResult();
+				}
+				association.setTranscriptAssociationSubject(parentTranscript);
+			}
+		}
+		association.setTranscriptCodingSequenceAssociationObject(cds);
+		association.setRelation(vocabularyTermService.getTermInVocabulary(VocabularyConstants.TRANSCRIPT_RELATION_VOCABULARY, VocabularyConstants.TRANSCRIPT_PARENT_TERM).getEntity());
+		
+		if (associationResponse.hasErrors()) {
+			throw new ObjectValidationException(gffEntry, associationResponse.errorMessagesString());
+		}
+
+		return transcriptCdsDAO.persist(association);
+	}
+	
+	@Transactional 
+	public TranscriptExonAssociation validateTranscriptExonAssociation(Gff3DTO gffEntry, Exon exon, Map<String, String> attributes) throws ObjectValidationException {
+		TranscriptExonAssociation association = new TranscriptExonAssociation();
+		
+		ObjectResponse<TranscriptExonAssociation> associationResponse = new ObjectResponse<TranscriptExonAssociation>();
+		if (!attributes.containsKey("Parent")) {
+			associationResponse.addErrorMessage("Attributes - Parent", ValidationConstants.REQUIRED_MESSAGE);
+		} else {
+			SearchResponse<Transcript> parentTranscriptSearch = transcriptDAO.findByField("modInternalId", attributes.get("Parent"));
+			if (parentTranscriptSearch == null || parentTranscriptSearch.getSingleResult() == null) {
+				associationResponse.addErrorMessage("Attributes - Parent", ValidationConstants.INVALID_MESSAGE + " (" + attributes.get("Parent") + ")");
+			} else {
+				Transcript parentTranscript = parentTranscriptSearch.getSingleResult();				Map<String, Object> params = new HashMap<>();
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.SingleSpaceSeparatorCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Use a single space to separate non-whitespace characters.
<!-- __reviewdog__:Cg8xN2IzOTZlMzYxMTM0NzUSCmNoZWNrc3R5bGU= -->